### DISCUSSION TITLE
Simple identifier not found only in JavaScript file

### DISCUSSION THREAD

**Discussion by instanceofme** [community/community#43980] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I was looking for a specific variable name, and the one instance I was looking for did not come up. The string was a simple `camelCaseIdentifier` (and searched as such in CS), in a repository private to my org. The results show the java and typescript instances of the identifier, but not the javascript one, even when I explicitly specified `repo:org/repo path:path/to/the/file.js` where it appears.

(Affected both the Code Search and the top search box from the repository in question.)

I've double-checked the identifier in question is present in the default (`master`) branch of said repository and has been there since 2019, so not new.

In the javascript file, it is referenced as `object.subobject.thatParticularIdentifier` (as well as in the `.ts` typescript fiels that does appear in the results too).

**Comment by look** [comment, marked as answer]
@instanceofme thanks for the report. Are you using the new Code Search beta? Are you able to perform other searches in this repository? What other results in the file?

I'm wondering if the repository or the file was excluded from the index (we list the reasons for exclusion here: https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations)

**Reply by instanceofme** [reply]
- The result in question did not show up in both Code Search beta and regular search from the repository's page top search bar (search in this repository)
- I tried searching another `camelCaseIdentifier` that appears in this file, same thing:
  - the result from this file does not appear
  - found in other files of the same repo, including `another.js` file in the same directory
  - applies both to CS beta & regular search
- The repository is definitely not excluded, I get other results
- I don't know the precise details of per-file exclusion but can say that:
  - the file is 247 KB, so less than 350 KB
  - it is encoded in UTF-8
  - it is not vendored nor generated
  - it is not ignored by `.gitignore` (nor in a path that would be, also see above, another .js file in that same directory gets results)

I can privately share the name of the private repository and path of the file to a GitHub employee if it may help

**Reply by look** [reply]
Sure, I can check it out. Please [file a bug report](https://support.github.com/contact/bug-report) and reference this discussion and ask that it be directed to @look. Thanks.

**Reply by instanceofme** [reply]
After checking with GitHub support, it turns out that we hit a different limit: 2 individual lines in that file were over 4096 bytes, which apparently also prevents indexing. We changed those lines to below this number, and it now appears in results.

**Reply by look** [reply]
Great to hear @instanceofme. The team is also working on improving this behavior.

**Reply by jsoref** [reply]
Reported as:
- [ ] https://github.com/github/docs/issues/31535
